core/test/utils/html_utils_test.dart

```diff
@@ -322,4 +322,381 @@ void main() {
       expect(HtmlUtils.extractPlainText(html), '<heloloasdadadadadsad>dab');
     });
   });
+
+  group('HtmlUtils.wrapPlainTextLinks', () {
```

Comment:
please try to add test cases
  - [ ] Protocol variations (ftp, mailto, file)
  - [ ] Fragment identifiers and combined query+fragment
  - [ ] URLs in code/pre tags (assume NOT wrapped)
  - [ ] JavaScript/data URLs (NOT wrapped for security)
  - [ ] URLs in attributes (NOT wrapped)
  - [ ] IP addresses
  - [ ] URLs in lists/tables/blockquotes
  - [ ] Malformed URLs
  - [ ] Adjacent links without spaces